#2766 Fixed Type Errors on System Settings

Patrykb0802 opened the pull request:
- Added new condition for save button in system settings to check if misc settings form is validated correctly
- Added isMiscValidated variable to handle validation
- Added createPath in UploadFileUtils.java, which prevents a situation where folder was created but files were saved in different location

Patrykb0802 committed:
#2766 Fixed Type Errors on System Settings

- Added new condition for save button in system settings to check if misc settings form is validated correctly
- Added isMiscValidated variable to handle validation
- Added createPath in UploadFileUtils.java, which prevents a situation where folder was created but files were saved in different location

Limraj commented on scadalts-ui/src/views/System/SystemSettings/MiscSettingsComponent.vue:
```diff
@@ -194,7 +195,7 @@ export default {
 				changed: changed,
 				data: this.sumarizeDataChanges(),
-        valid: this.validateForm(this.valid)
+				validateForm: this.validateForm(this.arePathsValid)
```
arePathsValid -> validatedPaths, validateForm -> valid

Patrykb0802 committed:
#2766 Fixed Type Errors on System Settings

- Added valid field to every settings component
- added additional validation checking if all components have passed validation (checkValidation())

Patrykb0802 committed:
#2766 Fixed Type Errors on System Settings

- fixed formatting

Patrykb0802 committed:
#2766 Fixed Type Errors on System Settings

Changed validation, so now it uses componentsEdited list

Limraj reviewed (approved):
1. A field validated with an error does not reset previously made changes;
2. Removing the cause of the validation error unlocks the save button and all previously introduced changes are visible and saved to the database;
